File: G171210045_3. Odev/pdp3/src/RastgeleKarakter.c
#include "RastgeleKarakter.h"

    rastgelekarakter rastgeleOlustur()
    {
        rastgelekarakter rg;    
        rg=(rastgelekarakter)malloc(sizeof(struct RastgeleKarakter));
        rg->rand=randomOlustur();
        rg->generateRandomChar=&GenerateRandomChar;
        rg->generateRandomCharBetween=&GenerateRandomCharBetween;
        rg->generateRandomChars=&GenerateRandomChars;
        rg->generateText=&GenerateText;
        rg->rastgeleyoket=&Rastgeleyoket;         
    }

    char* GenerateRandomCharBetween(const rastgelekarakter rg,char min, char max) 
    {        

        free(rg->currentarray);
        int rand=rg->rand->randomnumberbetween(rg->rand,(int)min,(int)max);
        rg->currentarray=(char*)malloc(sizeof(char)* rand);
        for(int i=0;i<rand;i++)
        {
            int a=rg->rand->randomnumberbetween(rg->rand,(int)min,(int)max);
            rg->currentarray[i]=(char)a;            
        }
        return rg->currentarray;
    }
    char GenerateRandomChar(const rastgelekarakter rg) 
    {
        int max=0;
        int min=0;
        int a=rg->rand->randomnumber(rg->rand);
        if(a%2==0)
        {
            max=90;         //büyük yada küçük
            min=65;
        }
        else
        {
            max=122;
            min=97;
        }
        int b=rg->rand->randomnumberbetween(rg->rand,min,max);
        char tmp=(char)b;
        return tmp;        
    }
    
    
    char* GenerateRandomChars(const rastgelekarakter rg,int value)
    {
        free(rg->currentarray);
        if(value>0)
        {
            rg->currentarray=(char*)malloc(sizeof(char)* value);
            for(int i=0;i<value;i++)                
                rg->currentarray[i]=GenerateRandomChar(rg);           
        }
        return rg->currentarray;        
    }
    
    char* GenerateText(const rastgelekarakter rg)            
    {
        free(rg->currentarray);
        free(rg->currenttext);
        int wordc=(rg->rand->randomnumberbetween(rg->rand,2,9));  //rastgele kelime sayısı
        int ac=0;
        rg->currenttext=(char*)malloc(sizeof (char) * 100);; 
        int tc=0;
        for(int i=0;i<wordc;i++)
        {
            ac=rg->rand->randomnumberbetween(rg->rand,2,9); //rastgele kelimenin harf sayısı
            for(int j=0;j<ac;j++)
            {
                rg->currenttext[tc]=GenerateRandomChar(rg);
                tc++;
            }
            rg->currenttext[tc]=' ';
            tc++;
        }
        return rg->currenttext;
    }

    void Rastgeleyoket(rastgelekarakter rg)
    {
        if(rg == NULL) return;
        rg->rand->randomyoket(rg->rand);
        free(rg->currentarray);
        free(rg->currenttext);
        free(rg);
        rg=NULL;
    }

File: G171210045_2.Odev/src/RastgeleKarakter/RastgeleKarakter.java
/**
 *
 *
 * @author Fatih Aslan
 *
 * @since 15.03.2019
 * <p>  
 *  No:G171210045
 *  RastgeleKarakterKütüphanesi
 * </p>
 */ 
package RastgeleKarakter;

public class RastgeleKarakter 
{
    int rd=(int)System.nanoTime()%10; //hızlı çalışmada nano time tekrar etmesin diye
    int generateRandomNumber(int min,int max)  //iki sayı arasında bir değeri rastgele üretiyor
    {           
        rd+=(int)System.nanoTime()%10;
        long a=System.nanoTime()*(System.nanoTime()%rd); //rastgeleliği kullanıcağımız değişken
        int tmp=(int)(a % (max+1));
        while(tmp<min)
        {
            tmp=(int)((System.nanoTime()) % (max+1));
        }
        return tmp;
    }
    public char[] generateRandomChar(char min, char max) 
    {
        int a=generateRandomNumber(1,3);
        char[] tmp=new char[a];
        for(int i=0;i<a;i++)
            tmp[i]=(char)generateRandomNumber((int)min,(int)max);
        return tmp;
    }
    public char generateRandomChar() 
    {
        int max=0;
        int min=0;
        if(System.nanoTime() %2==0)
        {
            max=90;         //büyük yada küçük
            min=65;
        }
        else
        {
            max=122;
            min=97;
        }
        char tmp=(char)generateRandomNumber(min,max);
        return tmp;        
    }
    
    
    public char[] generateRandomChar(int value)
    {
        if(value>0)
        {
            char[] txt=new char[value];  //sayı kadar char döner
            for(int i=0;i<value;i++)
                txt[i]=generateRandomChar();
            return txt;
        }
        return null;
    }
    
    public char[] takecharfrom(char... args) //girilen charlar arasından rastgele seçer
    {       
        int a=generateRandomNumber(1,2);
        char[] txt=new char[a];
        for (int i=0;i<a;i++)      
        {
            txt[i]=args[generateRandomNumber(0,args.length-1)];
        }
        return txt;
    }
    
    
    public char[] generateText()            
    {
        int wordc=generateRandomNumber(2,9);  //rastgele kelime sayısı
        int ac=0;
        char[] txt=new char[100]; 
        int tc=0;
        for(int i=0;i<wordc;i++)
        {
            ac=generateRandomNumber(2,9); //rastgele kelimenin harf sayısı
            for(int j=0;j<ac;j++)
            {
                txt[tc]=generateRandomChar();
                tc++;
            }
            txt[tc]=' ';
            tc++;
        }
        return txt;
    }
    
    
}
